src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigStorageLocation`1.cs

```diff
@@ -54,23 +54,30 @@ public bool TryGetOption(object underlyingOption, IReadOnlyDictionary<string, ob
         {
             if (allRawConventions.TryGetValue(KeyName, out object value))
             {
-                var optionalValue = _parseValue(value.ToString(), type);
-                if (optionalValue.HasValue)
-                {
-                    result = optionalValue.Value;
-                }
-                else
-                {
-                    result = null;
-                }
-
-                return result != null;
+                var ret = TryGetOption(value.ToString(), type, out var typedResult);
+                result = typedResult;
+                return ret;
             }
 
             result = null;
             return false;
         }
 
+        internal bool TryGetOption(string value, Type type, out T result)
+        {
+            var optionalValue = _parseValue(value, type);
```

Comment:
so, before, once option is accessed, its value was cached in OptionService. and later whole option was cached per document. so once option is accessed, ether getting option for document or getting value of the option is cheap.

now, with editor config, getting option value is no longer cached? making each access expensive? such as looping through all storages repeatedly or parsing option repeatedly each time the same option is accessed. 

probably we need to cache those values so that we don't repeatedly do same work?

Reply:
There is no behavior change in cfbc77ee1f0961916481e40301c81bc0f8c6f0e5. I simply extracted the method so a single-value dictionary didn't need to be wrapped in a dictionary object.

Reply:
sure. but regardless which PR did what,  shouldn't we open an issue to make sure we don't repeatedly do same work per each call? surprised that this didn't show up in trace.

Reply:
➡️ Filed #33328. I did see this in the trace, but it wasn't as severe an issue as the items fixed above. However, I didn't associate the cost with the lack of caching, so good catch here!